circleci: failed tests: TestClientNotReady

The following test appears to have failed:

[#20875](https://circleci.com/gh/cockroachdb/cockroach/20875):

```
I160802 15:48:15.936940 kv/dist_sender.go:950  application error: replay txn <nil>
--- PASS: TestSendNext_NonRetryableApplicationError (0.02s)
=== RUN   TestClientNotReady
W160802 15:48:15.960289 kv/dist_sender.go:966  RPC error: rpc error: code = 4 desc = context deadline exceeded
I160802 15:48:15.960804 /go/src/google.golang.org/grpc/clientconn.go:444  Failed to dial 127.0.0.1:48064: context canceled; please retry.
--- FAIL: TestClientNotReady (0.01s)
	send_test.go:447: grpc: the client connection is closing
=== RUN   TestComplexScenarios
W160802 15:48:15.968280 kv/dist_sender.go:966  RPC error: failed to send RPC: test
W160802 15:48:15.968393 kv/dist_sender.go:966  RPC error: failed to send RPC: test
W160802 15:48:15.968446 kv/dist_sender.go:966  RPC error: failed to send RPC: test
W160802 15:48:15.968501 kv/dist_sender.go:966  RPC error: failed to send RPC: test
W160802 15:48:15.968554 kv/dist_sender.go:966  RPC error: failed to send RPC: test
W160802 15:48:15.968634 kv/dist_sender.go:966  RPC error: failed to send RPC: test
W160802 15:48:15.968685 kv/dist_sender.go:966  RPC error: failed to send RPC: test
W160802 15:48:15.968771 kv/dist_sender.go:966  RPC error: failed to send RPC: test
--
I160802 15:49:44.496104 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
I160802 15:49:44.496492 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
--- PASS: TestRequestToUninitializedRange (0.33s)
=== RUN   TestForbiddenDeps
--- PASS: TestForbiddenDeps (22.00s)
FAIL
FAIL	github.com/cockroachdb/cockroach/kv	112.519s
=== RUN   TestCombinable
--- PASS: TestCombinable (0.00s)
=== RUN   TestMustSetInner
--- PASS: TestMustSetInner (0.00s)
=== RUN   TestBatchSplit
--- PASS: TestBatchSplit (0.00s)
=== RUN   TestBatchRequestGetArg
--- PASS: TestBatchRequestGetArg (0.00s)
=== RUN   TestBatchRequestString
--- PASS: TestBatchRequestString (0.00s)
--
I160802 15:52:19.520767 storage/replica.go:2610  gossiping sentinel from store 1, range 1
I160802 15:52:19.521057 storage/replica.go:2617  gossiping first range from store 1, range 1: [{1 1 1}]
I160802 15:52:19.523029 storage/replica.go:2610  gossiping sentinel from store 1, range 1
I160802 15:52:19.523420 storage/replica.go:2617  gossiping first range from store 1, range 1: [{1 1 1}]
==================
WARNING: DATA RACE
Write by goroutine 27:
  github.com/cockroachdb/cockroach/config.TestingSetDefaultZoneConfig()
      /go/src/github.com/cockroachdb/cockroach/config/config.go:82 +0xf8
  github.com/cockroachdb/cockroach/storage.TestSkipLargeReplicaSnapshot()
      /go/src/github.com/cockroachdb/cockroach/storage/replica_raftstorage_test.go:68 +0x202
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:473 +0xdc

Previous read by goroutine 65:
  github.com/cockroachdb/cockroach/config.SystemConfig.getZoneConfigForID()
      /go/src/github.com/cockroachdb/cockroach/config/config.go:294 +0x17a
  github.com/cockroachdb/cockroach/config.SystemConfig.GetZoneConfigForKey()
      /go/src/github.com/cockroachdb/cockroach/config/config.go:282 +0xe0
  github.com/cockroachdb/cockroach/storage.(*splitQueue).shouldQueue()
      /go/src/github.com/cockroachdb/cockroach/storage/split_queue.go:75 +0x1a7
  github.com/cockroachdb/cockroach/storage.(*baseQueue).MaybeAdd()
      /go/src/github.com/cockroachdb/cockroach/storage/queue.go:307 +0x424
  github.com/cockroachdb/cockroach/storage.(*Store).systemGossipUpdate()
      /go/src/github.com/cockroachdb/cockroach/storage/store.go:1290 +0x2fc
  github.com/cockroachdb/cockroach/storage.(*Store).Start.func4()
      /go/src/github.com/cockroachdb/cockroach/storage/store.go:1136 +0x135
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8a

Goroutine 27 (running) created at:
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:582 +0xae2
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:515 +0x11d
  github.com/cockroachdb/cockroach/storage_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/storage/main_test.go:57 +0x2d4
  main.main()
      github.com/cockroachdb/cockroach/storage/_test/_testmain.go:658 +0x20a

Goroutine 65 (running) created at:
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x6f
  github.com/cockroachdb/cockroach/storage.(*Store).Start()
      /go/src/github.com/cockroachdb/cockroach/storage/store.go:1141 +0x1634
  github.com/cockroachdb/cockroach/storage.createTestStoreWithContext()
      /go/src/github.com/cockroachdb/cockroach/storage/store_test.go:181 +0x263
  github.com/cockroachdb/cockroach/storage.TestSkipLargeReplicaSnapshot()
      /go/src/github.com/cockroachdb/cockroach/storage/replica_raftstorage_test.go:62 +0x152
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:473 +0xdc
--
=== RUN   Example_rebalancing
W160802 15:53:30.957723 gossip/gossip.go:1006  not connected to cluster; use --join to specify a connected node
--- PASS: Example_rebalancing (0.90s)
PASS
Found 1 data race(s)
FAIL	github.com/cockroachdb/cockroach/storage	75.786s
=== RUN   TestBatchBasics
I160802 15:53:44.230279 storage/engine/rocksdb.go:353  opening in memory rocksdb instance
I160802 15:53:44.233706 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.02s)
=== RUN   TestBatchRepr
I160802 15:53:44.253328 storage/engine/rocksdb.go:353  opening in memory rocksdb instance
I160802 15:53:44.255568 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
--- PASS: TestBatchRepr (0.03s)
=== RUN   TestBatchGet
I160802 15:53:44.269087 storage/engine/rocksdb.go:353  opening in memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/kv/send_test.go
+++ b/kv/send_test.go
@@ -443,7 +443,7 @@ func TestClientNotReady(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
-			if err := grpcConn.Close(); err != nil {
+			if err := grpcConn.Close(); err != nil && err != grpc.ErrClientConnClosing {
 				t.Fatal(err)
 			}
 
@@ -477,7 +477,7 @@ func TestClientNotReady(t *testing.T) {
 		_, err := sendBatch(SendOptions{
 			Context: context.Background(),
 		}, addrs, nodeContext)
-		if !testutils.IsError(err, "connection is closing") {
+		if !testutils.IsError(err, "connection is closing|failed fast due to transport failure") {
 			errCh <- errors.Wrap(err, "unexpected error")
 		} else {
 			close(errCh)
@@ -496,7 +496,7 @@ func TestClientNotReady(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	if err := grpcConn.Close(); err != nil {
+	if err := grpcConn.Close(); err != nil && err != grpc.ErrClientConnClosing {
 		t.Fatal(err)
 	}
 	for err := range errCh {

